notifier.go

```diff
@@ -102,37 +114,96 @@ func (n *SlackNotifier) notifyUser(c context.Context, userID, text string) error
 		return err
 	}
 
-	return n.notifyChannel(c, channel.ID, text)
+	_, err = n.notifyChannel(c, channel.ID, text)
+	return err
 }
 
-func (n *SlackNotifier) notifyChannel(c context.Context, channel, text string) error {
+func (n *SlackNotifier) notifyChannel(c context.Context, channel, text string) (timestamp string, err error) {
 	l := ctx.Logger(c).With("at", "slack.ping-user")
-
 	params := slackapi.NewPostMessageParameters()
 	params.AsUser = true
 	params.EscapeText = false
 
-	_, _, err := n.client.PostMessage(
+	var mainMsg slackapi.Blocks
+
+	// Main message
+	textBlock := slackapi.NewSectionBlock(
+		slackapi.NewTextBlockObject(slackapi.MarkdownType, text, false, false),
+		nil,
+		nil,
+	)
+
+	buttonBlock := slackapi.NewActionBlock(
+		"reviewer_response",
+		slackapi.NewButtonBlockElement("", reviewingRequestStatusMsg, slackapi.NewTextBlockObject("plain_text", "Looking", false, false)),
+		slackapi.NewButtonBlockElement("", unableToReviewStatusMsg, slackapi.NewTextBlockObject("plain_text", "Please reassign", false, false)),
+	)
+
+	mainMsg.BlockSet = append(mainMsg.BlockSet, textBlock)
+	mainMsg.BlockSet = append(mainMsg.BlockSet, buttonBlock)
+
+	// Send the main message requesting PR review
+	// Store the channel and timestamp of the message. We'll need this to update the message later
+	_, rts, err := n.client.PostMessageContext(
+		c,
 		channel,
-		slackapi.MsgOptionText(text, false),
+		slackapi.MsgOptionBlocks(mainMsg.BlockSet...),
 		slackapi.MsgOptionPostMessageParameters(params),
 	)
-	if err != nil {
-		l.Error("msg", "unable to post message", "err", err)
-		return err
+	l.Info("msg", "ping successful")
+	return rts, err
+}
+
+// Updates the original Slack message with a `context` block to show the status of the PR
+// See https://api.slack.com/reference/block-kit/blocks#context for clarification
+func (n *SlackNotifier) updateMessageWithContextBlock(ctx context.Context, payload *slackapi.InteractionCallback, text string) error {
```

Comment:
maybe something like this, as it always removes buttons?

```suggestion
func (n *SlackNotifier) replaceMessageButtonsWithNote(ctx context.Context, payload *slackapi.InteractionCallback, text string) error {
```